File: countRows.pl
#!/usr/bin/perl

use strict;
use warnings;

use Data::Dumper;

my $file=shift || die "usage: countRows.pl <filename>";

open(F,"<$file") || die "file $file not found";

my @count;
while(<F>) {
  chomp;
  my @a=split ',';
  for (my $i=0;  $i<scalar(@a);  $i++) {
    $count[$i]++ if (length ($a[$i])>0);
  }
}
close(F);

#print Dumper(\@count);

open(F,"<$file") || die "file $file not found";
while(<F>) {
  chomp;
  my @a=split ',';
  if (scalar(@a)>10) {
  for (my $i=0;  $i<scalar(@a);  $i++) {
    print "$a[$i]," if ($count[$i]>1600);   # should be 384*4+1
  }
  print "\n";
  }
}
close(F);

File: Java-Groovy-Perl/countRows.py
import sys

file = sys.argv[1] if len(sys.argv) > 1 else sys.exit("usage: countRows.py <filename>")

try:
    with open(file, "r") as f:
        lines = f.readlines()
except FileNotFoundError:
    print("File not found.")
    sys.exit(1)

counts = [0] * 1601  # 384*4+1
for line in lines:
    data = line.strip().split(",")
    for i in range(len(data)):
        if len(data[i]) > 0:
            counts[i] += 1

with open(file, "r") as f:
    lines = f.readlines()

for line in lines:
    data = line.strip().split(",")
    if len(data) > 10:
        for i in range(len(data)):
            if counts[i] > 1600:
                print(data[i], end=",")
        print()
